tests/realtikvtest/txntest/txn_test.go

```diff
@@ -251,3 +252,41 @@ func TestAssertionWhenPessimisticLockLost(t *testing.T) {
 	err := tk1.ExecToErr("commit")
 	require.NotContains(t, err.Error(), "assertion")
 }
+
+func TestSelectLockForPartitionTable(t *testing.T) {
+	store := realtikvtest.CreateMockStoreAndSetup(t)
+	tk1 := testkit.NewTestKit(t, store)
+	tk2 := testkit.NewTestKit(t, store)
+
+	tk1.MustExec("use test")
+	tk1.MustExec("create table t(a int, b int, c int, key idx(a, b, c)) PARTITION BY HASH (c) PARTITIONS 10")
+	tk1.MustExec("insert into t values (1, 1, 1)")
+	tk1.MustExec("analyze table t")
+	tk1.MustExec("begin")
+	tk1.HasPlan("select * from t use index(idx) where a = 1 and b = 1 for update", "IndexLookUp_9")
+	tk1.MustExec("select * from t use index(idx) where a = 1 and b = 1 for update")
+	ch := make(chan bool, 1)
```

Comment:
If you make it an unbuffered channel instead, you can use it as a semaphore/sync object, and skip the need for the sleep.
```suggestion
	ch := make(chan bool)
```

Reply:
Sorry for misunderstanding, you want the sleep to make sure the SelectLock starts to execute and blocks internally.